test/compiler.test.js

```diff
@@ -1014,6 +1014,201 @@ describe('compiler', function() {
+        });
+
+        expect(instructions.mixins).to.have.property('Vehicle');
+      });
+
+      it('resolves module relative path in `mixinSources` option', function() {
+        appdir.writeFileSync('node_modules/custom-mixins/vehicle.js', '');
+
+        var instructions = boot.compile({
+          appRootDir: appdir.PATH,
+          mixinSources: ['custom-mixins']
+        });
+
+        expect(instructions.mixins).to.have.property('Vehicle');
+      });
+
+      it('supports `mixins` option in model config', function() {
+        appdir.createConfigFilesSync({}, {}, {
+          _meta: {
+            mixins: ['mixins']
+          }
+        });
+
+        appdir.writeFileSync('mixins/other.js', '');
+        var instructions = boot.compile(appdir.PATH);
+
+        expect(instructions.mixins).to.have.property('Other');
+      });
+
+      it('resolves relative path in `mixins` option', function() {
+        appdir.createConfigFilesSync({}, {}, {
+          _meta: {
+            mixins: ['./custom-mixins']
+          }
+        });
+
+        appdir.writeFileSync('custom-mixins/vehicle.js', '');
+        var instructions = boot.compile(appdir.PATH);
+
+        expect(instructions.mixins).to.have.property('Vehicle');
+      });
+
+      it('resolves module relative path in `mixins` option', function() {
+        appdir.createConfigFilesSync({}, {}, {
+          _meta: {
+            mixins: ['custom-mixins']
+          }
+        });
+
+        appdir.writeFileSync('node_modules/custom-mixins/vehicle.js', '');
+        var instructions = boot.compile(appdir.PATH);
+
+        expect(instructions.mixins).to.have.property('Vehicle');
+      });
+
+      it('picks mixins from `mixinSources` when provided', function() {
+        appdir.writeFileSync('custom-mixins/other.js', '');
+        appdir.writeFileSync('mixins/default.js', '');
+
+        var instructions = boot.compile({
+          appRootDir: appdir.PATH,
+          mixinSources: ['custom-mixins']
+        });
+
+        expect(instructions.mixins).to.have.property('Other');
+        expect(instructions.mixins).to.not.have.property('Default');
+      });
+
+      it('supports mixin name normalization - classify', function() {
+        appdir.writeFileSync('mixins/foo.js', '');
+        appdir.writeFileSync('mixins/time-stamps.js', '');
+
+        var instructions = boot.compile(appdir.PATH);
+
+        expect(instructions.mixins).to.have.property('Foo');
+        expect(instructions.mixins).to.have.property('TimeStamps');
+      });
+
+      it('supports mixin name normalization - dasherize', function() {
+        appdir.writeFileSync('mixins/foo.js', '');
+        appdir.writeFileSync('mixins/time-stamps.js', '');
+
+        var options = { appRootDir: appdir.PATH, normalization: 'dasherize' };
+        var instructions = boot.compile(options);
+
+        expect(instructions.mixins).to.have.property('foo');
+        expect(instructions.mixins).to.have.property('time-stamps');
```

Comment:
Please check the following two file names too,  ideally in all "supports mixing name normalization" tests:

 - `mixins/camelCase.js`
 - `mixins/PascalCase.js`